tests/pkg/lwip/tests/01-run.py

```diff
@@ -196,7 +198,25 @@ def get_ipv6_address(spawn):
     return str(ipaddress.ip_address(spawn.match.group(1)))
 
 
-def test_ipv6_send(board_group, application, env=None):
+def get_ipv4_address(spawn):
+    ip_addr = "0.0.0.0"
+    for _ in range(5):  # wait for DHCP to configure address
+        spawn.sendline(u"ifconfig")
+        spawn.expect(r" inet addr: ((\d+\\.){3}\d+)\s")
```

Comment:
When this times out it will fail the test, it will not rerun the loop.

After DHCP exchange is complete lwIP will check with ARP a few times before considering the address as its own.

Reply:
> When this times out it will fail the test, it will not rerun the loop.

There will always be a parsable output. It will always show `0.0.0.0` which is tested below.

Reply:
> buffer (last 100 chars): 'nk: up State: up\r\n        Link type: wired\r\n        inet addr: 0.0.0.0 mask: 0.0.0.0 gw: 0.0.0.0\r\n> '
> before (last 100 chars): 'nk: up State: up\r\n        Link type: wired\r\n        inet addr: 0.0.0.0 mask: 0.0.0.0 gw: 0.0.0.0\r\n> '

Mbut it definitely shows up in your output :-/. Will try to reproduce.